examples/auth/app/auth/components/LoginForm.tsx

```diff
@@ -1,69 +1,79 @@
 import React from "react"
 import {LabeledTextField} from "app/components/LabeledTextField"
-import {Form} from "react-final-form"
-import {FORM_ERROR} from "final-form"
+import {FormProvider, useForm} from "react-hook-form"
 import login from "app/auth/mutations/login"
 
 type LoginFormProps = {
   onSuccess?: () => void
 }
 
 export const LoginForm = (props: LoginFormProps) => {
+  const methods = useForm({
+    mode: "onBlur",
+    async resolver(values) {
```

Comment:
Alternatively, if you are interested in steering users towards using more idiomatic JS:
```
async resolver(value) {
    return Object.keys(values).reduce((acc, key) => {
        if(requiredFields.includes(key) && !isRequiredValid(_values[key])) acc[key] = "Required"
        return acc;
    }, {})
}
```

Reply:
I find `reduce` very hard to read for working with objects :)

Reply:
I'd say it's more a case of getting used to it but you're not alone, for sure!